ghga_connector/cli.py

```diff
@@ -146,15 +153,38 @@ def download(  # noqa C901
     # perform the download:
     output_file = path.join(output_dir, file_id)
 
-    try:
-        download_file(download_url=download_url, output_file_path=output_file)
-    except BadResponseCodeError as error:
-        typer.echo(
-            "The download request was invalid and returnd a wrong HTTP status code."
-        )
-        raise error
-    except RequestFailedError as error:
-        typer.echo("The download request has failed.")
-        raise error
+    retries = 0
+    part_offset = 0
+
+    while part_offset < file_size:
+
+        # Calculetes end of byte range of the file
+        part_end = part_offset + part_size - 1
```

Comment:
Because otherwise we overshoot.

Example: We have a 16 bytes part size.
The first part is bytes 0 - 15

The Second part is byte 16 - 31

part_offset of the second part is 16

16 + 16 - 1 = 31